mono/eglib/glib.h

```diff
@@ -1123,6 +1123,12 @@ gboolean   g_file_test (const gchar *filename, GFileTest test);
 #else
 #define g_write write
 #endif
+#ifdef G_OS_WIN32
+#define g_read _read
+#else
+#define g_read read
+#endif
```

Comment:
On Unix `read` is a sycall, and therefore async-safe. On Windows it is not a syscall. So they are arguably quite differnent and maybe don't provide a portability layer over them, unless that portability layer is async-safe.

Reply:
Async-safe only if you preserve `errno` around use, that is.